app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -192,6 +192,10 @@ class DartdocJobProcessor extends JobProcessor {
           job.packageName, job.packageVersion);
       if (entry.isRegression(oldEntry)) {
         _logger.severe('Regression detected in $job, aborting upload.');
+        if (oldEntry.isLatest != entry.isLatest ||
+            oldEntry.isObsolete != entry.isObsolete) {
+          await dartdocBackend.updateOldEntry(oldEntry, entry);
```

Comment:
So, basically, if `isLatest` or `isObsolete` have changed, you still want to update the old entry, even if the job failed?

Please maybe add a one-line comment stating that..